src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VariantMongoWriter.java

```diff
@@ -81,4 +87,25 @@ private void executeBulk(BulkWriteOperation bulk, int currentBulkSize) {
         }
     }
 
+    private void generateIndexes() {
+        mongoOperations.getCollection(collection).createIndex(new BasicDBObject(
+                DBObjectToVariantConverter.CHROMOSOME_FIELD, 1)
+                                                                      .append(DBObjectToVariantConverter.START_FIELD, 1)
+                                                                      .append(DBObjectToVariantConverter.END_FIELD, 1)
+                                                                      .append("background", true));
+        mongoOperations.getCollection(collection).createIndex(new BasicDBObject(
+                DBObjectToVariantConverter.IDS_FIELD, 1).append("background", true));
+
+        String filesStudyIdField = String.format("%s.%s", DBObjectToVariantConverter.FILES_FIELD,
+                                                 DBObjectToVariantSourceEntryConverter.STUDYID_FIELD);
+        String filesFileIdField = String.format("%s.%s", DBObjectToVariantConverter.FILES_FIELD,
+                                                DBObjectToVariantSourceEntryConverter.FILEID_FIELD);
+        mongoOperations.getCollection(collection).createIndex(
+                new BasicDBObject(filesStudyIdField, 1).append(filesFileIdField, 1).append("background", true));
+        mongoOperations.getCollection(collection).createIndex(new BasicDBObject(
+                "annot.xrefs.id", 1).append("background", true));
+        mongoOperations.getCollection(collection).createIndex(new BasicDBObject("annot", 1).append("background", true));
+        
+        //// TODO: 05/12/2016 to create a unique index on { chr : 1, start : 1, end : 1, ref : 1, alt : 1 } double check with DBA
```

Comment:
It is not necessary, it was stated more to clarify that this comment no longer applied :)